### PR TITLE
Use driver without Venus OS

### PR DESCRIPTION
**Pull request by mr-manuel**
This PR will make it possible to use this driver also for other projects outside of Venus OS.

### CODE REVIEW

**Comment by mr-manuel** [comment]
@MacGH23 I opened a PR as draft, so we can discuss here on integration. Once you make changes and push them to your master branch it will update here automatically.

File: etc/dbus-serialbattery/uni_bms.py
Comment (mr-manuel) [inline etc/dbus-serialbattery/uni_bms.py:1]: This seems like a lot of duplicated code from `dbus-serialbattery.py`. Better would be to only extend the missing parts in `dbus-serialbattery.py` by extending the file or loading an additionall class into it.

File: etc/dbus-serialbattery/config.bmsonly.ini
[the remark this answers is not in the document]
Comment (MacGH23) [reply etc/dbus-serialbattery/config.bmsonly.ini:1]: The intension is to read the BMS systems with other python projects.
I just implementing a DIY Battery charging system for several devices for charging and discharging.
Here I wanted to use the info from the BMS to have more security.
I started to implement the reading from different BMS, but I found you already have almost all already implemented.
Therefore I tried to use your code without big modifications.
the uni-bms.py and bmshelper.py are the only 2 files needed currently.
An extra config file is not needed, I just disabled some settings with False as default.
All other is unchanged.

I already use this in my script with a raspberry pi and JKBMS. I read the values like voltages, current, ...
The usage is, for me, quite easy. in the test.py you can find the info.

File: etc/dbus-serialbattery/uni_bms.py
Comment (MacGH23) [reply etc/dbus-serialbattery/uni_bms.py:1]: yes thats correct, but I found no easy way to get it into dbushelper without many if:..else:
Therefore I use only the code I need to read the BMS.

File: etc/dbus-serialbattery/config.bmsonly.ini
[the remark this answers is not in the document]
Comment (MacGH23) [reply etc/dbus-serialbattery/config.bmsonly.ini:1]: I fyou like you can have a short look and my project, I send you an invitation

**Comment by mr-manuel** [comment]
Hello,

the driver had big changes. Please rebase your branch. You don't have to close and reopen this PR for doing this. After you made the rebase just force push the changes to your branch and it will update automatically this PR.

If it's easier for you to delete your fork/branch, create it from the upstream and apply your changes again, then you have to open a new PR.

Please add all changes with only one commit.

Take note, that also the license changed with v2.x.x.

**Comment by MacGH23** [comment]
OK, the normal BMS are already running, for can and modbus I need to understand the changes ;-)
When I'm finished I upload it

**Comment by MacGH23** [comment]
Question - Where do you create/bring up the canbus if not already initialized ?

**Comment by mr-manuel** [comment]
https://github.com/mr-manuel/venus-os_dbus-serialbattery/blob/0a2a09a75edb6248d8ee2847d71397c9549772ab/dbus-serialbattery/dbus-serialbattery.py#L358-L361

**Comment by MacGH23** [comment]
OK, misleading question ;-)

Here you open the can device, but you assume that the can0 is already available 
I search for craating the can0 if not available, like 
`
os.system('sudo slcand -f -s5 -o ' + self.CAN_DEVICE) #looks like a RS232 device, bring it up 
os.system('sudo ip link set can0 up type can bitrate 250000')
os.system('sudo ip link set up can0 txqueuelen 1000')
`

**Comment by mr-manuel** [comment]
This part is done by Venus OS.

**Comment by MacGH23** [comment]
Update uploaded.
I have no CAN or Bluetooth BMS to test, but should work

File: dbus-serialbattery/uni_bms.py [hunk 1]
```diff
@@ -0,0 +1,447 @@
+############################################################################
+#    Copyright (C) 2024 by macGH                                           #
+#                                                                          #
+#    This lib is free software; you can redistribute it and/or modify      #
+#    it under the terms of the LGPL                                        #
```
Comment (mr-manuel) [inline dbus-serialbattery/uni_bms.py:5]: Version 2 of this software is licensed under the terms of AGPL. Therefore you have to license is also as AGPL, remove this note that the main license applies or stick with the version before v2.

Comment (MacGH23) [reply dbus-serialbattery/uni_bms.py]: Ah forgot it, I refer to the LICENSE file now

**Comment by mr-manuel** [comment]
I merge it now as it is. I will then make some code and name changes.

**Comment by MacGH23** [comment]
Thanks

**Pull request merged**